### PR TITLE
ARUHA-554 Introduce zero-padded offsets

### PR DESCRIPTION
**Pull request by antban**

### CODE REVIEW

File: src/main/java/org/zalando/nakadi/service/CursorConverter.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    public Cursor convert(final NakadiCursor nakadiCursor) {
+        final boolean zeroPaddedOffsets =
+                featureToggleService.isFeatureEnabled(FeatureToggleService.Feature.ZERO_PADDED_OFFSETS);
+        final String offset;
+        if (nakadiCursor.getOffset().equals("-1")) {
+            offset = Cursor.BEFORE_OLDEST_OFFSET;
+        } else {
+            if (zeroPaddedOffsets) {
+                offset = StringUtils.leftPad(nakadiCursor.getOffset(), 18, '0');
```
Comment (lmontrieux) [inline src/main/java/org/zalando/nakadi/service/CursorConverter.java]: Maybe use a constant for `18`?

File: src/main/java/org/zalando/nakadi/service/CursorConverter.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.zalando.nakadi.domain.NakadiCursor;
+import org.zalando.nakadi.util.FeatureToggleService;
+import org.zalando.nakadi.view.Cursor;
+import org.zalando.nakadi.view.SubscriptionCursor;
+
+@Service
+public class CursorConverter {
```
Comment (v-stepanov) [inline src/main/java/org/zalando/nakadi/service/CursorConverter.java]: Why this class doesn't do conversion back to NakadiCursor?

Comment (antban) [inline src/main/java/org/zalando/nakadi/service/CursorConverter.java]: @v-stepanov As we discussed, it will be better to implement it a little bit later

**Comment by v-stepanov** [comment]
👍

**Comment by v-stepanov** [comment]
:shipit:

**Comment by v-stepanov** [comment]
👍

**Comment by v-stepanov** [comment]
👍

**Comment by v-stepanov** [comment]
👍

**Comment by v-stepanov** [comment]
👍

**Comment by v-stepanov** [comment]
:shipit:

**Comment by v-stepanov** [comment]
I will try to close this one and to open another one.

**Comment by antban** [comment]
As zappr doesn't want to work for this PR , I'll close this one and open another one.

**Comment by antban** [comment]
Here it is: https://github.com/zalando/nakadi/pull/568

**Pull request closed without merging**